src/open-uri.c

```diff
@@ -919,6 +919,27 @@ handle_open_in_thread_func (GTask *task,
                                                      g_object_ref (request));
 }
 
+static gboolean
+handle_scheme_supported (XdpDbusOpenURI *object,
+                         GDBusMethodInvocation *invocation,
+                         const gchar *arg_scheme)
+{
+  if (arg_scheme == NULL || *arg_scheme == '\0')
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             XDG_DESKTOP_PORTAL_ERROR,
+                                             XDG_DESKTOP_PORTAL_ERROR_INVALID_ARGUMENT,
+                                             "Scheme not specified");
+      return G_DBUS_METHOD_INVOCATION_HANDLED;
+    }
+  GAppInfo* app_info = g_app_info_get_default_for_uri_scheme (arg_scheme);
```

Comment:
You are leaking `GAppInfo`. 
The docs says: The caller of the function takes ownership of the data, and is responsible for freeing it.